Index: pcsx2-0.9.8/pcsx2/gui/AppConfig.cpp
===================================================================
--- pcsx2-0.9.8.orig/pcsx2/gui/AppConfig.cpp	2011-04-10 16:10:07.000000000 +0200
+++ pcsx2-0.9.8/pcsx2/gui/AppConfig.cpp	2011-04-10 16:12:23.000000000 +0200
@@ -133,7 +133,12 @@
 	{
 		switch( mode )
 		{
+#ifdef __LINUX__
+            // By default on linux move all user data file into central configuration directory
+			case DocsFolder_User:	return GetUserLocalDataDir();
+#else
 			case DocsFolder_User:	return (wxDirName)Path::Combine( wxStandardPaths::Get().GetDocumentsDir(), pxGetAppName() );
+#endif
 			case DocsFolder_Custom: return CustomDocumentsFolder;
 
 			jNO_DEFAULT
